service/worker/scanner/history/scavenger.go

```diff
@@ -262,6 +262,8 @@ func (s *Scavenger) startTaskProcessor(
 					}
 					domainName, err := s.domainCache.GetDomainName(task.domainID)
 					if err != nil {
+						s.logger.Error("Not able to fetch domain name",
+							getTaskLoggingTags(err, task)...)
 						return
```

Comment:
we need to send the error to the channel, and also do `continue` instead of `return`.
See how we handle other errors in this func.